Lib/argparse.py

```diff
@@ -659,6 +663,82 @@ def _split_lines(self, text, width):
         return text.splitlines()
 
 
+class FlexiHelpFormatter(HelpFormatter):
+    """Help message formatter which respects paragraphs and bulleted lists.
+
+    Only the name of this class is considered a public API. All the methods
+    provided by the class are considered an implementation detail.
+    """
+
+    def _split_lines(self, text, width):
+        return self._para_reformat(text, width)
+
+    def _fill_text(self, text, width, indent):
+        lines = self._para_reformat(text, width)
+        return "\n".join(lines)
```

Comment:
```suggestion
        lines = self._para_reformat(text, width - len(indent))
        return "\n".join(indent + line for line in lines)
```

As is, this eats any form of indentation that the user might have specified, which I think we want to preserve.

Reply:
Commit 2a53ad49b3.